src/Route/RouteCollection.php

```diff
@@ -83,7 +83,8 @@ public function add(
         $routeName = $this->baseRouteName.'_'.$name;
 
         if (!isset($defaults['_controller'])) {
-            $defaults['_controller'] = $this->baseControllerName.':'.$this->actionify($code);
+            $actionJoiner = false === \strpos($this->baseControllerName, '\\') ? ':' : '::';
```

Comment:
Should we trigger a deprecation here? We are probably not going to support the old notation on next major…

Reply:
Unluckily I don't know the deprecation policy of the project. But it would be nice to be rid of this silliness.

Reply:
@sonata-project/contributors please advise

Reply:
IMO we could maybe merge as is and add the deprecation later if really need be

Reply:
Deprecate for sure, at one point we will just use `::`, right?

Reply:
Correct!

Reply:
hm, true, then skip ti.

Reply:
Yes skip it

Reply:
It seems this change causes a BC break. In my admin service configuration I've put as baseControllerName value `AppBundle:Admin\Info` 

Shouldn't we also check for the `:` presence ?